File: dsa-interview-crash-course/0724_find_pivot_index.cpp
#include <vector>

class Solution {
   public:
    int pivotIndex(std::vector<int>& nums) {
        int prev = 0;
        for (int i = 0; i < nums.size(); ++i) {
            nums[i] += prev;
            prev = nums[i];
        }

        prev = 0;
        for (int i = 0; i < nums.size(); ++i) {
            int post = nums.back() - nums[i];
            if (prev == post) return i;
            prev = nums[i];
        }
        return -1;
    }
};

File: leetcode-75/0724_find_pivot_index.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        prefixSum = [0] * (len(nums) + 2)
        for i in range(1, len(nums) + 1):
            prefixSum[i] = prefixSum[i - 1] + nums[i - 1]

        for i in range(1, len(nums) + 1):
            leftside = prefixSum[i - 1]
            rightside = prefixSum[-2] - prefixSum[i]
            if leftside == rightside:
                return i - 1

        return -1